### PR TITLE
Add prefixes to overlapping widgets/classes

### LINKED ISSUE #107: Renaming widgets with "macos" prefix
All our widget names should start with Mac. eg `MacScaffold`.
This is to prevent name conflicts with existing widgets in the flutter framework and preserve the great developer experience Flutter already has. No dev (yes, including us) dreams of importing our library with a prefix or in every file of their project.

### PR DESCRIPTION
**Pull request by GroovinChip**
This PR contains version 0.3.0 of `macos_ui`, which contains the following breaking changes:
* Add `MacosPrefix` to widgets/classes with names that overlap with the material/cupertino libraries:
  * `TextField` -> `MacosTextField`
  * `Scaffold` -> `MacosTextField`
  * `IconButton` -> `MacosIconButton`
  * `BackButton -> `MacosBackButton`
  * `Scrollbar` -> `MacosScrollbar`
  * `Checkbox` -> `MacosCheckbox`
  * `RadioButton` -> `MacosRadioButton`
  * `Tooltip` -> `MacosTooltip`
  * `Typography` -> `MacosTypography`
  * `Switch` -> `MacosSwitch`

Closes #107 

## Pre-launch Checklist

- [x] I have run `dartfmt` on all changed files
- [x] I have run "optimize/organize imports" on all changed files
- [x] I have addressed all analyzer warnings as best I could
- [x] I have added/updated relevant documentation
- [x] I have incremented the package version as appropriate and updated `CHANGELOG.md` with my changes
- [x] I have run `flutter pub publish --dry-run` and addressed any warnings

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -9,15 +9,17 @@ Flutter widgets and themes implementing the current macOS design language.
   - [Contributing](#contributing)
   - [Resources](#resources)
 - [Layout](#layout)
-  - [Scaffold](#scaffold)
+  - [MacosScaffold](#macosscaffold)
 - [Buttons](#buttons)
-  - [Checkbox](#checkbox)
+  - [MacosCheckbox](#macoscheckbox)
   - [HelpButton](#helpbutton)
```
Comment (bdlukaa) [inline README.md:15]: MacosHelpButton, MacosRadioButton, MacosPushButton

**Review by bdlukaa — DISMISSED** [review]

File: README.md (hunk 1 above)
Comment (GroovinChip) [reply README.md:15]: They don't overlap with any pre-existing widgets. Not needed.

File: CHANGELOG.md [hunk 2]
```diff
@@ -1,3 +1,16 @@
+## [0.3.0]
+* Add `MacosPrefix` to widgets/classes with names that overlap with the material/cupertino libraries:
+  * `TextField` -> `MacosTextField`
+  * `Scaffold` -> `MacosTextField`
+  * `IconButton` -> `MacosIconButton`
+  * `BackButton -> `MacosBackButton`
```
Comment (bdlukaa) [inline CHANGELOG.md]: missing `

**Review by bdlukaa — APPROVED** [review]
Widgets such as indicators (progress, capacity and rating) should also be renamed.

File: CHANGELOG.md (hunk 2 above)
Comment (GroovinChip) [reply CHANGELOG.md]: Whoops, I'll fix that

**Comment by GroovinChip** [comment]
> Widgets such as indicators (progress, capacity and rating) should also be renamed.

No, I checked and they don't overlap

**Pull request merged**